tests/components/doorbird/test_config_flow.py

```diff
@@ -0,0 +1,258 @@
+"""Test the DoorBird config flow."""
+import urllib
+
+from asynctest import MagicMock, patch
+
+from homeassistant import config_entries, data_entry_flow, setup
+from homeassistant.components.doorbird import CONF_CUSTOM_URL, CONF_TOKEN
+from homeassistant.components.doorbird.const import CONF_EVENTS, DOMAIN
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+
+from tests.common import MockConfigEntry, init_recorder_component
+
+VALID_CONFIG = {
+    CONF_HOST: "1.2.3.4",
+    CONF_USERNAME: "friend",
+    CONF_PASSWORD: "password",
+    CONF_NAME: "mydoorbird",
+}
+
+
+def _get_mock_doorbirdapi_return_values(ready=None, info=None):
+    doorbirdapi_mock = MagicMock()
+    type(doorbirdapi_mock).ready = MagicMock(return_value=ready)
+    type(doorbirdapi_mock).info = MagicMock(return_value=info)
+
+    return doorbirdapi_mock
+
+
+def _get_mock_doorbirdapi_side_effects(ready=None, info=None):
+    doorbirdapi_mock = MagicMock()
+    type(doorbirdapi_mock).ready = MagicMock(side_effect=ready)
+    type(doorbirdapi_mock).info = MagicMock(side_effect=info)
+
+    return doorbirdapi_mock
+
+
+async def test_user_form(hass):
+    """Test we get the user form."""
+    await hass.async_add_job(init_recorder_component, hass)  # force in memory db
```

Comment:
We should use `hass.async_add_executor_job` instead of `hass.async_add_job`. The latter is legacy for general use.

Reply:
This is copy pasta.  I'll clean it up everywhere.

Reply:
https://github.com/home-assistant/core/pull/33185